src/liquid/any.cr

```diff
@@ -259,10 +259,14 @@ module Liquid
       Any.new(-raw)
     end
 
+    # Checks that the underlying value is a `Number`, and returns its value.
+    # Raises otherwise.
     def as_number : Number
-      as_number? || 0
+      as_number? || raise TypeCastError.new("Cast from String to Number+ failed")
```

Comment:
I think the error message could just tell it's not a number, because an Any isn't always a String as the message suggests.

Reply:
Good idea. I updated the message.